### ISSUE TITLE
File sharing extension breaks login for some users

### ISSUE THREAD

**Issue by agilob** [nextcloud/server#2740]
When some users of my system try to login using browser or desktop app they get an error. The logs I see in data/nextcloud.log is:

> "reqId":"Cg09egxNrM3SD3xkErZX","remoteAddr":"IP_ADDR","app":"index","message":"Exception: {\"Exception\":\"InvalidArgumentException\",\"Message\":\"\",\"Code\":0,\"Trace\":\"#0 \\\NEXTCLOUD/lib\\\/private\\\/Share20\\\/DefaultShareProvider.php(758): OC\\\\Share20\\\\Share->setSharedBy(NULL)\\n#1 \\\NEXTCLOUD/lib\\\/private\\\/Share20\\\/DefaultShareProvider.php(622): OC\\\\Share20\\\\DefaultShareProvider->createShare(Array)\\n#2 \\\NEXTCLOUD/lib\\\/private\\\/Share20\\\/Manager.php(957): OC\\\\Share20\\\\DefaultShareProvider->getSharedWith('USER', 0, NULL, -1, 0)\\n#3 \\\NEXTCLOUD/apps\\\/files_sharing\\\/lib\\\/MountProvider.php(71): OC\\\\Share20\\\\Manager->getSharedWith('USER', 0, NULL, -1)\\n#4 \\\NEXTCLOUD/lib\\\/private\\\/Files\\\/Config\\\/MountProviderCollection.php(77): OCA\\\\Files_Sharing\\\\MountProvider->getMountsForUser(Object(OC\\\\User\\\\User), Object(OC\\\\Files\\\\Storage\\\\StorageFactory))\\n#5 [internal function]: OC\\\\Files\\\\Config\\\\MountProviderCollection->OC\\\\Files\\\\Config\\\\{closure}(Object(OCA\\\\Files_Sharing\\\\MountProvider))\\n#6 \\\NEXTCLOUD/lib\\\/private\\\/Files\\\/Config\\\/MountProviderCollection.php(78): array_map(Object(Closure), Array)\\n#7 \\\NEXTCLOUD/lib\\\/private\\\/Files\\\/Filesystem.php(426): OC\\\\Files\\\\Config\\\\MountProviderCollection->getMountsForUser(Object(OC\\\\User\\\\User))\\n#8 \\\NEXTCLOUD/lib\\\/private\\\/Files\\\/Filesystem.php(374): OC\\\\Files\\\\Filesystem::initMountPoints('USER')\\n#9 \\\NEXTCLOUD/lib\\\/private\\\/legacy\\\/util.php(227): OC\\\\Files\\\\Filesystem::init('USER', '\\\/USER\\\/files')\\n#10 \\\NEXTCLOUD/lib\\\/base.php(982): OC_Util::setupFS()\\n#11 \\\NEXTCLOUD/index.php(48): OC::handleRequest()\\n#12 {main}\",\"File\":\"\\\NEXTCLOUD/lib\\\/private\\\/Share20\\\/Share.php\",\"Line\":288}","level":3,"time":"2016-12-18T20:58:33+00:00","method":"GET","url":"\/index.php\/apps\/gallery\/config?extramediatypes=1","user":"USER","version":"9.1.2.2"}
> reqId":"tcXwwuWEXYJQEfxhBKJw","remoteAddr":"IP_ADDR","app":"index","message":"Exception: {\"Exception\":\"InvalidArgumentException\",\"Message\":\"\",\"Code\":0,\"Trace\":\"#0 \\\NEXTCLOUD/lib\\\/private\\\/Share20\\\/DefaultShareProvider.php(758): OC\\\\Share20\\\\Share->setSharedBy(NULL)\\n#1 \\\NEXTCLOUD/lib\\\/private\\\/Share20\\\/DefaultShareProvider.php(622): OC\\\\Share20\\\\DefaultShareProvider->createShare(Array)\\n#2 \\\NEXTCLOUD/lib\\\/private\\\/Share20\\\/Manager.php(957): OC\\\\Share20\\\\DefaultShareProvider->getSharedWith('USER', 0, NULL, -1, 0)\\n#3 \\\NEXTCLOUD/apps\\\/files_sharing\\\/lib\\\/MountProvider.php(71): OC\\\\Share20\\\\Manager->getSharedWith('USER', 0, NULL, -1)\\n#4 \\\NEXTCLOUD/lib\\\/private\\\/Files\\\/Config\\\/MountProviderCollection.php(77): OCA\\\\Files_Sharing\\\\MountProvider->getMountsForUser(Object(OC\\\\User\\\\User), Object(OC\\\\Files\\\\Storage\\\\StorageFactory))\\n#5 [internal function]: OC\\\\Files\\\\Config\\\\MountProviderCollection->OC\\\\Files\\\\Config\\\\{closure}(Object(OCA\\\\Files_Sharing\\\\MountProvider))\\n#6 \\\NEXTCLOUD/lib\\\/private\\\/Files\\\/Config\\\/MountProviderCollection.php(78): array_map(Object(Closure), Array)\\n#7 \\\NEXTCLOUD/lib\\\/private\\\/Files\\\/Filesystem.php(426): OC\\\\Files\\\\Config\\\\MountProviderCollection->getMountsForUser(Object(OC\\\\User\\\\User))\\n#8 \\\NEXTCLOUD/lib\\\/private\\\/Files\\\/Filesystem.php(374): OC\\\\Files\\\\Filesystem::initMountPoints('USER')\\n#9 \\\NEXTCLOUD/lib\\\/private\\\/legacy\\\/util.php(227): OC\\\\Files\\\\Filesystem::init('USER', '\\\/USER\\\/files')\\n#10 \\\NEXTCLOUD/lib\\\/base.php(982): OC_Util::setupFS()\\n#11 \\\NEXTCLOUD/index.php(48): OC::handleRequest()\\n#12 {main}\",\"File\":\"\\\NEXTCLOUD/lib\\\/private\\\/Share20\\\/Share.php\",\"Line\":288}","level":3,"time":"2016-12-18T20:58:46+00:00","method":"GET","url":"\/index.php\/apps\/files\/","user":"USER","version":"9.1.2.2"}
> {"reqId":"Ui6j7c3piofI6yLd3ktm","remoteAddr":"IP_ADDR","app":"webdav","message":"Exception: {\"Message\":\"HTTP\\\/1.1 503 InvalidArgumentException: \",\"Exception\":\"Sabre\\\\DAV\\\\Exception\\\\ServiceUnavailable\",\"Code\":0,\"Trace\":\"#0 \\\NEXTCLOUD/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Auth\\\/Plugin.php(166): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\Auth->check(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#1 [internal function]: Sabre\\\\DAV\\\\Auth\\\\Plugin->beforeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#2 \\\NEXTCLOUD/3rdparty\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#3 \\\NEXTCLOUD/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(446): Sabre\\\\Event\\\\EventEmitter->emit('beforeMethod', Array)\\n#4 \\\NEXTCLOUD/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(248): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#5 \\\NEXTCLOUD/apps\\\/dav\\\/appinfo\\\/v1\\\/webdav.php(60): Sabre\\\\DAV\\\\Server->exec()\\n#6 \\\NEXTCLOUD/remote.php(165): require_once('\\\NEXTCLOUD/...')\\n#7 {main}\",\"File\":\"\\\NEXTCLOUD/apps\\\/dav\\\/lib\\\/Connector\\\/Sabre\\\/Auth.php\",\"Line\":162,\"User\":\"USER\"}","level":4,"time":"2016-12-18T20:58:46+00:00","method":"PROPFIND","url":"\/remote.php\/webdav\/","user":"USER","version":"9.1.2.2"}
{"reqId":"r81QuCUc+j1+hul2UAGq","remoteAddr":"IP_ADDR","app":"index","message":"Exception: {\"Exception\":\"InvalidArgumentException\",\"Message\":\"\",\"Code\":0,\"Trace\":\"#0 \\\NEXTCLOUD/lib\\\/private\\\/Share20\\\/DefaultShareProvider.php(758): OC\\\\Share20\\\\Share->setSharedBy(NULL)\\n#1 \\\NEXTCLOUD/lib\\\/private\\\/Share20\\\/DefaultShareProvider.php(622): OC\\\\Share20\\\\DefaultShareProvider->createShare(Array)\\n#2 \\\NEXTCLOUD/lib\\\/private\\\/Share20\\\/Manager.php(957): OC\\\\Share20\\\\DefaultShareProvider->getSharedWith('USER', 0, NULL, -1, 0)\\n#3 \\\NEXTCLOUD/apps\\\/files_sharing\\\/lib\\\/MountProvider.php(71): OC\\\\Share20\\\\Manager->getSharedWith('USER', 0, NULL, -1)\\n#4 \\\NEXTCLOUD/lib\\\/private\\\/Files\\\/Config\\\/MountProviderCollection.php(77): OCA\\\\Files_Sharing\\\\MountProvider->getMountsForUser(Object(OC\\\\User\\\\User), Object(OC\\\\Files\\\\Storage\\\\StorageFactory))\\n#5 [internal function]: OC\\\\Files\\\\Config\\\\MountProviderCollection->OC\\\\Files\\\\Config\\\\{closure}(Object(OCA\\\\Files_Sharing\\\\MountProvider))\\n#6 \\\NEXTCLOUD/lib\\\/private\\\/Files\\\/Config\\\/MountProviderCollection.php(78): array_map(Object(Closure), Array)\\n#7 \\\NEXTCLOUD/lib\\\/private\\\/Files\\\/Filesystem.php(426): OC\\\\Files\\\\Config\\\\MountProviderCollection->getMountsForUser(Object(OC\\\\User\\\\User))\\n#8 \\\NEXTCLOUD/lib\\\/private\\\/Files\\\/Filesystem.php(374): OC\\\\Files\\\\Filesystem::initMountPoints('USER')\\n#9 \\\NEXTCLOUD/lib\\\/private\\\/legacy\\\/util.php(227): OC\\\\Files\\\\Filesystem::init('USER', '\\\/USER\\\/files')\\n#10 \\\NEXTCLOUD/lib\\\/base.php(982): OC_Util::setupFS()\\n#11 \\\NEXTCLOUD/index.php(48): OC::handleRequest()\\n#12 {main}\",\"File\":\"\\\NEXTCLOUD/lib\\\/private\\\/Share20\\\/Share.php\",\"Line\":288}","level":3,"time":"2016-12-18T20:58:47+00:00","method":"GET","url":"\/index.php\/apps\/gallery\/config?extramediatypes=1","user":"USER","version":"9.1.2.2"}

Disabling the extension makes things work.

**Comment by schiessle** [comment]
Just a rough guess, but it seems like we try to mount a share which don't exists. @rullzer any idea

**Comment by rullzer** [comment]
Yes seems that way. I'll go trough the backtrace.

**Comment by rullzer** [comment]
@agilob please do fill out the whole issue template https://github.com/nextcloud/server/blob/master/issue_template.md

**Comment by agilob** [comment]
### Expected behaviour
Should not cause login/show files error, should be able to share files too.

### Actual behaviour
Login using a browser or desktop client causes error, unable to list files of some users.

### Server configuration
**Operating system**: debian 

**Web server:** nginx

**Database:** mysql

**PHP version:*
~/ $ php -v
PHP 5.6.28-0+deb8u1 (cli) (built: Nov 15 2016 06:01:37) 

**Nextcloud version:** (see Nextcloud admin page) Latest stable

**Updated from an older Nextcloud/ownCloud or fresh install:** updated from ownCloud

**Where did you install Nextcloud from:** official .zip 


**List of activated apps:**
```
$sudo -u nginx php occ app:list
sudo: unable to resolve host queennode
Enabled:
  - admin_audit: 1.0.0
  - calendar: 1.4.1
  - contacts: 1.5.2
  - dav: 1.0.1
  - documents: 0.13.1
  - federatedfilesharing: 1.0.1
  - files: 1.5.2
  - files_pdfviewer: 0.8.1
  - files_texteditor: 2.1
  - files_videoplayer: 0.9.8
  - gallery: 15.0.1
  - password_policy: 1.0.0
  - provisioning_api: 1.0.0
  - serverinfo: 1.1.1
  - survey_client: 0.1.5
  - tasks: 0.9.4
  - updatenotification: 1.0.1
  - workflowengine: 1.0.1
Disabled:
  - activity
  - comments
  - encryption
  - external
  - federation
  - files_accesscontrol
  - files_automatedtagging
  - files_external
  - files_retention
  - files_sharing
  - files_trashbin
  - files_versions
  - firstrunwizard
  - notifications
  - systemtags
  - templateeditor
  - theming
  - user_external
  - user_ldap
  - user_saml
```
**The content of config/config.php:**
```
{
    "system": {
        "instanceid": "oci4ro44d792",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "***REMOVED SENSITIVE VALUE***",
            "***REMOVED SENSITIVE VALUE***",
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "overwrite.cli.url": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "9.1.2.2",
        "dbname": "owncloud",
        "dbhost": "localhost",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "forcessl": true,
        "forceSSLforSubdomains": true,
        "theme": "",
        "maintenance": false,
        "ldapIgnoreNamingRules": false,
        "appstore.experimental.enabled": true,
        "loglevel": 0,
        "trashbin_retention_obligation": "auto",
        "mail_from_address": "owncloud",
        "mail_smtpmode": "php",
        "mail_domain": "agilob.net",
        "updater.secret": "***REMOVED SENSITIVE VALUE***"
    }
}
```

As stated above, the problem is caused by file_sharing plugin, disabling it fixes issues for some user, other users complain they can't share files externally.

**Comment by MorrisJobke** [comment]
@schiessle @rullzer Happened again in https://github.com/nextcloud/server/issues/4698

**Comment by agilob** [comment]
I just upgraded to 11.0.3 and no longer have this issue.

**Comment by MorrisJobke** [comment]
@jplouvel is this also fixed for you in 11.0.3?
